File: boca-1.5.1/doc/tests/fontes/bits-runtime.pas
program bits;
var i, inst, max, n: Longint;
    v: array[0..1] of Longint;
begin
  while true do
  begin
    read(n);
    if n=0 then break;
    max := 1;
    for i:=0 to n-1 do
    begin
      read(v[i]);
      if v[i]>max then max:=v[i]
    end;
    i:=0;
    while max>0 do
    begin
      max := max div 2;
      i := i + 1
    end;
    write('Instancia ');
    inst := inst + 1;
    writeln(inst);
    writeln(i*n);
    writeln()
  end
end.

File: boca-1.5.2/doc/tests/fontes/bits-runtime.java
import java.io.*;
class bits {
  public static void main(String args[]) throws Exception {
    StreamTokenizer st = new StreamTokenizer(System.in);
    int i, j, n, max, inst=0;
    while(42==42) {
      st.nextToken();
      n = (int) st.nval;
      if(n==0) break;
      max=1;
      for(i=0; i<n; i++) {
        st.nextToken();
        j = (int) st.nval;
        if(j>max) max=j;
      }
      for(i=0; max>0; i++)
        max >>= 1;
      inst++;
      System.out.println("Instancia " + inst);
      System.out.println(i*n + "\n");
      throw new Exception("OPA! Runtime error, hehehe...");
    }
  }
}
